tests/drivers/stepper/stepper_api/src/main.c

```diff
@@ -122,87 +141,257 @@ ZTEST_F(stepper, test_actual_position)
+	/* Run the stepper in positive direction */
+	(void)stepper_run(fixture->dev, STEPPER_DIRECTION_POSITIVE);
 
-	if (ret == -ENOSYS) {
-		ztest_test_skip();
+	/* Stop the stepper */
+	int ret = stepper_stop(fixture->dev);
+	bool is_moving;
+
+	if (ret == 0) {
+		POLL_AND_CHECK_SIGNAL(stepper_signal, stepper_event, STEPPER_EVENT_STOPPED,
+				      K_NO_WAIT);
+		zassert_equal(user_data_received, fixture->dev, "User data not received");
+
+		/* Check if the stepper is stopped */
+		stepper_is_moving(fixture->dev, &is_moving);
+		zassert_equal(is_moving, false, "Stepper is still moving");
+	} else if (ret == -ENOSYS) {
+		stepper_is_moving(fixture->dev, &is_moving);
+		zassert_equal(is_moving, true,
+			      "Stepper should be moving since stop is not implemented");
+	} else {
+		zassert_unreachable("Stepper stop failed");
 	}
-	/* Pass the function name as user data */
+}
+
+ZTEST_F(stepper, test_move_to_positive_direction_movement)
+{
+	int32_t pos = 20u;
+
+	Z_TEST_SKIP_IFDEF(CONFIG_STEPPER_TEST_ACCELERATION);
+
+	(void)stepper_enable(fixture->dev);
+	(void)stepper_set_microstep_interval(fixture->dev, TEST_STEP_INTERVAL);
 	(void)stepper_set_event_callback(fixture->dev, fixture->callback, (void *)fixture->dev);
+	(void)stepper_move_to(fixture->dev, pos);
+
+	POLL_AND_CHECK_SIGNAL(
+		stepper_signal, stepper_event, STEPPER_EVENT_STEPS_COMPLETED,
+		K_MSEC(pos * (20 + CONFIG_STEPPER_TEST_TIMING_TIMEOUT_TOLERANCE_PCT)));
 
+	(void)stepper_get_actual_position(fixture->dev, &pos);
+	zassert_equal(pos, 20u, "Target position should be %d but is %d", 20u, pos);
+	zassert_equal(user_data_received, fixture->dev, "User data not received");
+}
+
+ZTEST_F(stepper, test_move_to_negative_direction_movement)
+{
+	int32_t pos = -20u;
+
+	Z_TEST_SKIP_IFDEF(CONFIG_STEPPER_TEST_ACCELERATION);
+
+	(void)stepper_enable(fixture->dev);
+	(void)stepper_set_microstep_interval(fixture->dev, TEST_STEP_INTERVAL);
+	(void)stepper_set_event_callback(fixture->dev, fixture->callback, (void *)fixture->dev);
 	(void)stepper_move_to(fixture->dev, pos);
 
 	POLL_AND_CHECK_SIGNAL(
 		stepper_signal, stepper_event, STEPPER_EVENT_STEPS_COMPLETED,
-		K_MSEC(pos * (100 + CONFIG_STEPPER_TEST_TIMING_TIMEOUT_TOLERANCE_PCT)));
+		K_MSEC(-pos * (20 + CONFIG_STEPPER_TEST_TIMING_TIMEOUT_TOLERANCE_PCT)));
 
 	(void)stepper_get_actual_position(fixture->dev, &pos);
-	zassert_equal(pos, 10u, "Target position should be %d but is %d", 10u, pos);
+	zassert_equal(pos, -20u, "Target position should be %d but is %d", -20u, pos);
 	zassert_equal(user_data_received, fixture->dev, "User data not received");
 }
 
+ZTEST_F(stepper, test_move_to_is_moving_false_when_completed)
+{
+	int32_t pos = 20;
+	bool moving;
+
+	Z_TEST_SKIP_IFDEF(CONFIG_STEPPER_TEST_ACCELERATION);
+
+	(void)stepper_enable(fixture->dev);
+	(void)stepper_set_microstep_interval(fixture->dev, TEST_STEP_INTERVAL);
+	(void)stepper_set_event_callback(fixture->dev, fixture->callback, NULL);
```

Comment:
> Also, before somebody asks, I removed the LOG_DBG in stepper_print_event_callback because it would cause native_sim to crash.

LOG_DBG in itself is not causing the native_sim to crash. Passing a NULL as user_data  here is causing the crash with stepper_print_event_callback
```suggestion
        (void)stepper_set_event_callback(fixture->dev, fixture->callback, (void *)fixture->dev);
```

Reply:
> Also, before somebody asks, I removed the LOG_DBG in stepper_print_event_callback because it would cause native_sim to crash.

@jbehrensnx It's better to triage something locally before making bold statements like this. 
I was actually confused as to how `LOG_DBG` doesn't run in your case, where I keep using it when running tests and assumed this is some new changes with the testing framework.

Reply:
I meant the specific `LOG_DBG` statement, not `LOG_DBG` in general, sorry if that wasn't clear.
I probably should have done further experiments why the statement caused the error, but at that point I was just relieved to have found the source of the issue (I canceled a previous attempt at test generalization because I couldn't find the source of this issue). Commenting it out and doing further research would have been the better solution.

Reply:
>  I removed the LOG_DBG in stepper_print_event_callback because it would cause native_sim to crash.

The crash should have happened on HW as well, because a NULL was getting passed around.

It's okay, sometimes things can be overlooked while developing, that's we get our work reviewed by each other and that's also the collaborative spirit of Open Source and Zephyr Project :). In general, I think we need to be conscious about the selection of words, for instance, this statement in the PR description, can come across as not so positive.
> While several stepper drivers have been developed, testing was often done only haphazardly, primarily using the stepper shell, such as PR 79120.

I highly appreciate the side effects which this PR have produced from introducing stepper_stop to improving API specification. Thanks for that :)

In general, there are Code Of Conduct guidelines in ZephyrProject https://github.com/zephyrproject-rtos/zephyr/blob/main/CODE_OF_CONDUCT.md#our-standards and we should abide by them.